cli/src/pcluster/validators/instances_validators.py

```diff
@@ -243,11 +249,28 @@ def _validate(
         memory_scheduling_enabled: bool,
         **kwargs,
     ):
-        """Memory-based scheduling is NOT supported for Compute Resources with multiple instance types."""
-        if memory_scheduling_enabled and len(instance_types_info.items()) > 1:
+        if (
+            memory_scheduling_enabled
+            and len(instance_types_info.items()) > 1
+            and self.memory_difference_exceeds_thresholds(instance_types_info)
+        ):
             self._add_failure(
-                "Memory-based scheduling is only supported for Compute Resources using either 'InstanceType' or "
-                f"'Instances' with one instance type. Compute Resource {compute_resource_name} has more than "
-                "one instance type specified.",
-                FailureLevel.ERROR,
+                f'Enabling Memory-based scheduling when a Compute Resource ("{compute_resource_name}") has more than'
+                " one instance type specified may lead to unused resources since only the minimum available memory"
+                " across all instance-types can be specified in the Slurm node definition.",
+                FailureLevel.WARNING,
             )
+
+    def memory_difference_exceeds_thresholds(self, instance_types_info):
+        """Return True only if both thresholds are exceeded."""
+        min_memory, max_memory = self._min_max_memory(instance_types_info)
+        absolute_diff = max_memory - min_memory
+        percent_diff = absolute_diff / min_memory
+
+        return absolute_diff > MIN_MEMORY_ABSOLUTE_DIFFERENCE and percent_diff > MIN_MEMORY_PRECENTAGE_DIFFERENCE
```

Comment:
Nitpick: maybe `>=` instead of `>`?

Reply:
It's a threshold. 
I think a strict check it's better.